custom-resource-handlers/src/pgp-secret.ts

```diff
@@ -153,3 +153,11 @@ async function _getPublicKey(secretArn: string): Promise<string> {
     await _rmrf(tempDir);
   }
 }
+
+async function _deleteSecret(event: cfn.DeleteEvent): Promise<cfn.ResourceAttributes> {
```

Comment:
I believe @RomainMuller intentionally omitted this (although not sure why)

Reply:
The reason I added it was because without it, we need to manually delete the secret between the runs of our integration tests (`npm run test update`) using the AWS CLI (the console does not allow immediately deleting a secret). Note that the default for this resource is `RemovalPolicy.RETAIN`.

If we need it, I think it's reasonable to expect that other customers of `delivlib` will need it too.

@RomainMuller are you OK with adding this delete capability to `OpenPGPKeyPair`?

Reply:
I had left it out because:
- You should NOT take the private key out of the AWS Account (the point is that no human ever has access to that)
- This implies you may not have a backup of it outside of the cloud
- If you inadvertently lose it for whatever reason, you might be in a very difficult spot.

If we implement `Delete` actions, we __MUST__ not force immediate deletion (or at least force the customer to very explicitly opt into that behavior). It's way too risky.

Reply:
How about something like this?

```ts
interface OpenPGPKeyPairProps {
 // other properties as today...

  /** @default OpenPGPKeyRemovalPolicy.RETAIN */
  readonly removalPolicy?: OpenPGPKeyRemovalPolicy;
}

export enum OpenPGPKeyRemovalPolicy {
  RETAIN, DESTROY_SAFELY, DESTROY_IMMEDIATELY
}
```
```

Reply:
I'd be okay with that.

Reply:
Done in the newest revision.